streams/src/main/java/org/apache/kafka/streams/kstream/internals/CogroupedKStreamImpl.java

```diff
@@ -0,0 +1,195 @@
+                                                        final StateStoreSupplier storeSupplier,
+                                                        final Merger<? super K, V> sessionMerger,
+                                                        final SessionWindows sessionWindows,
+                                                        final Windows<W> windows) {
+        final Set<String> sourceNodes = new HashSet<>();
+        final Collection<KStreamAggProcessorSupplier> processors = new ArrayList<>();
+        final List<String> processorNames = new ArrayList<>();
+        for (final Map.Entry<KGroupedStream, Aggregator> pair : pairs.entrySet()) {
+            final KGroupedStreamImpl groupedStream = (KGroupedStreamImpl) pair.getKey();
+            final String sourceName = repartitionIfRequired(groupedStream);
+            if (sourceName.equals(groupedStream.name)) {
+                sourceNodes.addAll(groupedStream.sourceNodes);
+            } else {
+                sourceNodes.add(sourceName);
+            }
+
+            final KStreamAggProcessorSupplier processor;
+            switch (aggregateType) {
+                case AGGREGATE:
+                    processor = new KStreamAggregate(storeSupplier.name(), initializer, pair.getValue());
+                    break;
+                case SESSION_WINDOW_AGGREGATE:
+                    processor = new KStreamSessionWindowAggregate(sessionWindows, storeSupplier.name(), initializer, pair.getValue(), sessionMerger);
+                    break;
+                case WINDOW_AGGREGATE:
+                    processor = new KStreamWindowAggregate(windows, storeSupplier.name(), initializer, pair.getValue());
+                    break;
+                default:
+                    throw new IllegalStateException("Unrecognized AggregateType.");
+            }
+            processors.add(processor);
+            
+            final String processorName = topology.newName(COGROUP_AGGREGATE_NAME);
+            processorNames.add(processorName);
+            topology.addProcessor(processorName, processor, sourceName);
+        }
+        final String name = topology.newName(COGROUP_NAME);
+        final KStreamCogroup cogroup = new KStreamCogroup(processors);
+        final String[] processorNamesArray = processorNames.toArray(new String[processorNames.size()]);
+        topology.addProcessor(name, cogroup, processorNamesArray);
+        topology.addStateStore(storeSupplier, processorNamesArray);
+        topology.copartitionSources(sourceNodes);
+        return new KTableImpl<>(topology, name, cogroup, sourceNodes, storeSupplier.name());
+    }
+
+    @SuppressWarnings("rawtypes")
+    private String repartitionIfRequired(KGroupedStreamImpl groupedStream) {
+        if (repartitionNames.containsKey(groupedStream)) {
+            return repartitionNames.get(groupedStream);
+        }
+        final String sourceName = groupedStream.repartitionIfRequired(null);
```

Comment:
This is not in this PR: I realized that in `KGroupedStreamsImpl#repartitionIfRequired`, in the return statement:

```
return KStreamImpl.createReparitionedSource(this, keySerde, valSerde, queryableStoreName);
```

We pass the `queryableStoreName` as the prefix of the repartition topic. That seems not correct to me? cc @enothereska since it seems from one of your previous commits,.